fastscanner/adapters/realtime/redis_channel.py

```diff
@@ -48,10 +48,15 @@ async def flush(self):
             await pipeline.execute()
 
     async def subscribe(self, channel_id: str, handler: ChannelHandler) -> None:
-        self._handlers.setdefault(channel_id, []).append(handler)
+        if channel_id not in self._handlers:
+            self._handlers[channel_id] = []
+
+        if handler not in self._handlers[channel_id]:
+            self._handlers[channel_id].append(handler)
+
```

Comment:
```suggestion
     self._handlers.setdefault(channel_id, []).append(handler)
```
This line was okay, the problem is that you were appending again in line 54.